### PR TITLE
refactor: convert config API to async await

### PR DESCRIPTION
**Pull request by alanshaw**
BREAKING CHANGE: Errors returned from request failures are now all [`HTTPError`](https://github.com/sindresorhus/ky/blob/c0d9d2bb07e4c122a08f019b39e9c55a4c9324f3/index.js#L117-L123)s which carry a `response` property. This is a [`Response`](https://developer.mozilla.org/en-US/docs/Web/API/Response) that can be used to inspect _all_ information relating to the HTTP response. This means that the `err.status` or `err.statusCode` property should now be accessed via `err.response.status`.

### CODE REVIEW

File: src/config/set.js [hunk 1]
```diff
@@ -1,35 +1,36 @@
...
 
-module.exports = (send) => {
-  return promisify((key, value, opts, callback) => {
-    if (typeof opts === 'function') {
-      callback = opts
-      opts = {}
-    }
     if (typeof key !== 'string') {
-      return callback(new Error('Invalid key type'))
+      throw new Error('invalid key')
```
Comment (achingbrain) [inline src/config/set.js]: Why the change in message?  Is this for go compatibility or similar?

File: test/lib.error-handler.spec.js [hunk 2]
```diff
@@ -21,8 +21,9 @@ describe('lib/error-handler', () => {
 
     const err = await throwsAsync(errorHandler(null, null, res))
 
+    expect(err instanceof HTTPError).to.be.true()
     expect(err.message).to.eql('boom')
-    expect(err.status).to.eql(500)
+    expect(err.response.status).to.eql(500)
```
Comment (achingbrain) [inline test/lib.error-handler.spec.js]: Breaking change?  I guess errors aren't really part of the spec other than 'you will get an error'.

Comment (alanshaw) [reply test/lib.error-handler.spec.js]: Yes, it is breaking. I've updated the PR desctiption with the info and will include it when merging.

File: src/config/set.js (hunk 1 above)
Comment (alanshaw) [reply src/config/set.js]: No, I will revert.

File: test/lib.error-handler.spec.js (hunk 2 above)
Comment (achingbrain) [reply test/lib.error-handler.spec.js]: Great.  One small thing - the description says `err.status` has changed but it's `err.statusCode`

Comment (alanshaw) [reply test/lib.error-handler.spec.js]: Well this test was looking for `.status` but others are looking for `.statusCode`...I've added both to the breaking change message.
